library/std/src/lib.rs

```diff
@@ -174,7 +174,9 @@
 //!
 //! - after-main use of thread-locals, which also affects additional features:
 //!   - [`thread::current()`]
-//! - before-main stdio file descriptors are not guaranteed to be open on unix platforms
+//! - under UNIX, before main, file descriptors 0, 1, and 2 are unchanged
```

Comment:
I don't think we should guarantee that the stdio file descriptors are unchanged before main. That would forbid attempting to open the as /dev/null before main through for example a global constructor.

Reply:
You're right, I downgraded this to "may be".